### DISCUSSION TITLE
Thank you! and then two suggestions ;-)

### DISCUSSION THREAD

**Discussion by linuxnewbee** [M-Reimer/savescreenshot#48] [Ideas]
Hi, thanks for this extension! I like it.
I would also like to do two suggestions to make it even more useful:
1) Selection -> Store x and y coordinates (so I can repeat the screen shot with same size, similar to Greenshot for Windows: Capture last region)
2) Add to the selection list: Repeat last action (last selected alternative like x, y and format)
Makes it very easy to repeat something you just did.

**Comment by linuxnewbee** [comment]
In fact I have a third suggestion too:
Some times the page is too long even for screenshot (lack of resources), then I would like it to take the remaining in a second shot. Currently I have to take the rest of the page with a different screen saver :-o

**Comment by M-Reimer** [comment]
At first: Thanks for your feedback.

About the "last selection" suggestion: Where exactly would this be useful? I can't think of any case.

About the "lack of resources" part: Why do you need such long screenshots? Probably you use the wrong tool for the job. Please do not waste space by archiving websites as images! Probably what you really need is an Add-on which is able to save PDF. Like this one: https://addons.mozilla.org/de/firefox/addon/print-edit-we/

**Comment by linuxnewbee** [comment]
Thanks, that was quick ;-)

2) I frequently "save the weather", almost daily, from this site:
https://en.ilmatieteenlaitos.fi/local-weather/helsinki
There you see the local weather with 7 tabs: Weather, Wind, Humidity....
I then use "Selection (PNG)", but have to repeat the same procedure for each tab.
Much more convenient would be to select the first tab as before and then switch tab and click "capture last region" as I do in Windows using Greenshot (not working with Linux).
There may be other better ways to do this, but my programming-skills in for instance Python are not that developed yet.

3) Yes but...
I also visit/join online auctions having a long list of items. Mostly they fit on one page, but not always.
The auction goes quickly at the end so there is not much time for fiddling with tools, your screenshot works perfectly and takes a snapshot exactly when I want. I will try your suggestion but so far I have been using only your screenshot and afterwards I see that something was lost or not.
When the "stressy" auction is over I can peacefully go back and delete the long pages after looking/picking the interesting parts. My intention is not to save the web ;-)

**Comment by linuxnewbee** [comment]
Hi again, 
I tried and failed with print-edit-we. Playing with options did not change anything for me.
No images included and far from WYSIWYG while your tool looks and works perfect.
The only issue I have is the Firefox maximum pixels limitation.
For instance this site:
https://www.slag1.se/auction/avyttring-av-per-larssons-traktormuseum-i-staffanstorp/

Thanks, 
Bo

**Comment by M-Reimer** [comment]
> No images included and far from WYSIWYG while your tool looks and works perfect.
> The only issue I have is the Firefox maximum pixels limitation.

But there is the maximum pixel limitation where I can (and will) do nothing about. Simply out of scope for a screenshot tool which has the goal to be simple and easy.

> For instance this site:
> https://www.slag1.se/auction/avyttring-av-per-larssons-traktormuseum-i-staffanstorp/

Wow. That's honestly the biggest pile of garbage I have ever seen. Someone should tell them that HTML has an &lt;img&gt; tag to place images. They used a &lt;div&gt; and added the image using 'style="background: url(...)"'.

And then they add this to their print stylesheet which basically breaks all images for printout even with the regular browser print feature:

```
@media print {
  * {
    background: transparent !important;
```

But back to topic: This can be fixed with "Print Edit WE" if you click the "Web Style" button in the toolbar. After enabling this, choose "Save" -> "... PDF" and don't forget to tick "Show images".

**Comment by M-Reimer** [comment]
I'll have a look at the "Last selection" feature. This should be relatively easy to implement as long as it is a "per tab" feature (means that last selection can not be shared between tabs).

**Comment by linuxnewbee** [comment]
> Wow. That's honestly the biggest pile of garbage I have ever seen.

He, he ;-) 
It's probably a museum going down.
Remember, old tech was the start of new tech once and so it continues. New tech rarely pops up from nowhere even if it happens, mostly it's a long process full of smaller steps making it gradually "better". Even some proud German stuff there ;-)

No, I understand the pixel limitation issue is not your thing. I was only asking if it is possible to repeat with a new "full page" starting at where the last was cut, one way or the other.

If not, I have to continue using the very heavy and "clumpsy" to use Awsome screenshot (with several bugs and too many clicks) for the missing part. One good thing with that screen saver is that you can very easily select freely over long pages, unless that is also possible with your fine tool one day (suggestion 4) ;-)

Thanks for the hint, now I also got images. However, a 49Mb file, where you make a 4.9Mb file + the remaining end, maybe 1Mb max. I think I will continue with your tool.

And yes, your tool is very simple and easy to use, so 10 points from Finland.

**Comment by Gitoffthelawn** [comment]
> If not, I have to continue using the very heavy and "clumpsy" to use Awsome screenshot (with several bugs and too many clicks) for the missing part. One good thing with that screen saver is that you can very easily select freely over long pages, unless that is also possible with your fine tool one day (suggestion 4) ;-)

Following this thread... by "Awsome screenshot", are you referring to this?:   
https://addons.mozilla.org/firefox/addon/screenshot-capture-annotate/ 

Occasionally, I too, have very long web pages I want to save as an image, so it will be good to learn if that extension is the one that works for that task.

**Comment by linuxnewbee** [comment]
Yes, that's the one.
Works, but with surprises and takes some time getting used to.
I think it takes any long size page, but typically I only use it for the last remaining part and when I need to select an area over several browser pages where it excels.

One other type of long pages I have is from a remote RPi web-cam.
https://elinux.org/RPi-Cam-Web-Interface
Highly recommended and very user customizable/flexible. Only drawback is very long web-pages, specially windy weather in sunshine with lot's of false positives. Can be fixed with external IR-detectors though.

**Reply by Gitoffthelawn** [reply]
Thank you for the feedback and additional ideas!

**Comment by M-Reimer** [comment]
> Remember, old tech was the start of new tech once and so it continues.

I think &lt;img&gt; was part of the very first HTML standard. So no excuse to not use it.

> Thanks for the hint, now I also got images. However, a 49Mb file, where you make a 4.9Mb file + the remaining end, maybe 1Mb max. I think I will continue with your tool.

OK. So they even fail with this. You always have at least two resolutions. One thumbnail resolution for fast loading and one big resolution if a user wants to see details. They always use the big images...

But probably time to learn something about common Linux tools:
https://gist.github.com/firstdoit/6390547
At least on my distribution the command "ghostscript" does not exist. Probably you have to replace this with just "gs" like I had to do. My final command:

    $ gs -sDEVICE=pdfwrite -dCompatibilityLevel=1.4 -dPDFSETTINGS=/screen -dNOPAUSE -dQUIET -dBATCH -sOutputFile=output.pdf Nätauktioner\ -\ Slag1.pdf

Result:

    $ ls -lh *.pdf
    -rw-r--r-- 1 user group  48M  3. Jan 14:25 'Nätauktioner - Slag1.pdf'
    -rw-r--r-- 1 user group 3,7M  3. Jan 14:28  output.pdf

**Reply by linuxnewbee** [reply]
Thanks again for your excellent support!

Works great, 3.7Mb, the full page, also for me.
Maybe I have to consider using these tools, at least for "static" auctions ;-)

**Comment by M-Reimer** [comment]
Moved the whole thread to a discussion, now. I'll set up issues to cover individual points and link them here.

**Comment by M-Reimer** [comment]
New Issues created from this discussion are:
#49 and #50 
Please add comments, relevant to an individual Issue, to the individual Issue and feel free to continue general discussion here.

**Comment by M-Reimer** [comment]
Version 9.3.0 with the features, discussed here, is on its way.

**Reply by linuxnewbee** [reply]
Thanks, this version was more useful than I first thought.
I can now easily cut the ugly top-banner by scrolling down before selecting "full page".

One more interesting example of the "copy with last settings" web:
https://www.svk.se/en/national-grid/the-control-room/
Here you can graphically see the Nordic countries power production and consumption in real time with the possibility to go back many days.
Scroll down to the middle where you have a colorful chart, select country and date.
You will be surprised how unreliable wind power is when we have the coldest weather and thus the biggest need.
Look at Finland today "2022-02-08" and compare few days earlier when there was more wind. Cutting the nuclear power is very risky and will cause high prices of electricity as we just realized. Then compare with Norway having plenty of hydro power.
Interesting data! It will affect your own power bill.
